initial-promises/node-server/inventory/1.0/fusionAgent.cf

```diff
@@ -451,6 +453,24 @@ bundle agent addInformationsToInventory
         edit_defaults => def_no_backup;
 }
 
+# Check that te inventory file is valid
+bundle agent checkInventoryFile
+{
+  vars:
+      "inventory_file" slist => findfiles("${g.rudder_var_tmp}/inventory/*.ocs");
+  classes:
+      "file_exist" expression => filesexist("@{inventory_file}");
+    file_exist::
+      "inventory_valid" expression => returnszero("${this.promise_dirname}/test-inventory.pl ${inventory_file}"),
+                        scope => "namespace";
+
+  reports:
+    file_present.!inventory_valid::
+      "@@Inventory@@result_error@@inventory-all@@inventory-all@@00@@inventory@@None@@${g.execRun}##${g.uuid}@#Generated inventory has been detected as invalid";
+      # Print information to the user in addition to the reporting
+      "WARNING: your inventory is not valid, not sending it!";
```

Comment:
I think it should display in a more visible way like what we do for the update error
**********************************************************
* Generated inventory is not valid, as it is missing
* mandatory fields.
* Not sending it to the Rudder Server
* You can check the invalid inventory in ${g.rudder_var_tmp}/inventory/
*********************************************************************************************

or something like that
If it were sent at the end of the run, it is even better